ui/src/app/views/recent-work/component.ts

```diff
@@ -14,21 +23,57 @@ import {
 export class RecentWorkComponent implements OnInit {
   resourceList: RecentResource[];
   fullList: RecentResource[] = [];
+  @Input('workspace')
+  workspace: Workspace;
+  accessLevel: WorkspaceAccessLevel;
   startIndex = 0;
+  cssClass: string;
+  resourcesLoading = false;
   constructor(
-    private userMetricsService: UserMetricsService
+    private userMetricsService: UserMetricsService,
+    private workspacesService: WorkspacesService,
+    private cohortsService: CohortsService,
+    private route: ActivatedRoute,
   ) {}
   index: Number;
   size = 3;
+
   ngOnInit(): void {
+    this.resourcesLoading = true;
     this.updateList();
   }
 
   updateList(): void {
-    this.userMetricsService.getUserRecentResources().subscribe((resources) => {
-      this.fullList = resources;
-      this.resourceList = this.fullList.slice(this.startIndex, this.startIndex + this.size);
-    });
+    if (this.workspace) {
+      this.cssClass = 'for-list';
+      const wsData: WorkspaceData = this.route.snapshot.data.workspace;
+      this.accessLevel = wsData.accessLevel;
+      const notebookCall = this.workspacesService
+        .getNoteBookList(this.workspace.namespace, this.workspace.id);
+      const cohortCall = this.cohortsService
+        .getCohortsInWorkspace(this.workspace.namespace, this.workspace.id);
+      Observable.forkJoin(notebookCall, cohortCall)
+        .subscribe(([notebooks, cohorts]) => {
+          const notebookResources = convertToResources(notebooks, this.workspace.namespace,
+            this.workspace.id, this.accessLevel, ResourceType.NOTEBOOK);
+          const cohortResources = convertToResources(cohorts.items, this.workspace.namespace,
+            this.workspace.id, this.accessLevel, ResourceType.COHORT);
+          this.fullList = notebookResources.concat(cohortResources);
+          this.fullList.sort((leftSide, rightSide): number => {
+            if (leftSide.modifiedTime < rightSide.modifiedTime) { return 1; }
+            if (leftSide.modifiedTime > rightSide.modifiedTime) { return -1; }
```

Comment:
opt_nit: Personally I prefer either braces OR 1-line if statements without braces. Optional since we haven't really decided on a team style convention on this point.

Reply:
Yeah at some point when we're not quite so under the gun, I think we should have a dev meeting (or two) on style conventions and do up a style guide... I'll put it on my list of things to think about after Algood?